### PR TITLE
New swapchain model

### LINKED ISSUE #322: Use the new swapchain model of gfx-rs
We need to port our swapchain code to gfx-rs as of https://github.com/gfx-rs/gfx/pull/2882 in order to improve resizing support. Basing on the current gfx-rs master also has the benefit of a lighter build. It does, however, make it a conflict with #320 and will prevent us from using hotfixes in gfx-rs.

The prerequisite of this issue is updating Rendy - https://github.com/amethyst/rendy/issues/194

### PR DESCRIPTION
**Pull request by kvark**
Fixes #322
TODO:
- [x] Test on Metal
- [x] Test on DX12
- [x] Test on Vulkan
- [x] Try embedding the backend information into `SwapChainId`
- [x] Wait for https://github.com/amethyst/rendy/pull/202

### CODE REVIEW

**Comment by kvark** [comment]
@grovesNL this is all functional and ready for the review!

File: wgpu-native/src/swap_chain.rs [hunk 1]
```diff
@@ -133,221 +114,120 @@ impl SwapChainDescriptor {
 #[repr(C)]
 #[derive(Debug)]
 pub struct SwapChainOutput {
-    pub texture_id: TextureId,
+    //pub texture_id: TextureId,
```
Comment (grovesNL) [inline wgpu-native/src/swap_chain.rs]: nit: leftover comment

File: wgpu-native/src/swap_chain.rs [hunk 2]
```diff
@@ -1,81 +1,62 @@
+/*! Swap chain management.
+
+    ## Lifecycle
+
+    At the low level, the swap chain is using the new simplified model of gfx-rs.
```
Comment (grovesNL) [inline wgpu-native/src/swap_chain.rs]: Thanks for the explanation, this is really helpful!

**Comment by kvark** [comment]
@grovesNL thanks for the review!
I rebased the code and updated the description of what's actually done.
Note that the biggest unknown here is whether just having a `TextureView` for the swapchain image is enough... I'll be talking about this on the upcoming F2F. If that's not the case, we'd have to go back to gfx-rs and extend its new swapchain model, which is not trivial.

Anyhow, let's proceed 🚀 
bors r=grovesNL

**Pull request merged**